src/runtime/auto-prompt-manager.ts

```diff
@@ -550,21 +552,39 @@ export class AutoPromptManager {
     autoPromptType?: AutoPromptType;
     isClosable?: boolean;
   }): () => void {
-    return () => {
-      const params: AudienceActionIframeParams = {
-        action,
-        configurationId,
-        autoPromptType,
-        onCancel: () => this.storeLastDismissal_(),
-        isClosable,
+    if (action == TYPE_REWARDED_ADS) {
+      return () => {
+        const lastAudienceActionFlow = new AudienceActionLocalFlow(this.deps_, {
+          action,
+          configurationId,
+          autoPromptType,
+          onCancel: this.storeLastDismissal_.bind(this),
+          isClosable,
+          monetizationFunction: this.getMonetizationPromptFn_(
+            autoPromptType,
+            !!isClosable
+          ),
+        });
+        this.setLastAudienceActionFlow(lastAudienceActionFlow);
+        lastAudienceActionFlow.start();
       };
-      const lastAudienceActionFlow = new AudienceActionIframeFlow(
-        this.deps_,
-        params
-      );
-      this.setLastAudienceActionFlow(lastAudienceActionFlow);
-      lastAudienceActionFlow.start();
-    };
+    } else {
+      return () => {
+        const params: AudienceActionIframeParams = {
+          action,
+          configurationId,
+          autoPromptType,
+          onCancel: () => this.storeLastDismissal_(),
+          isClosable,
+        };
+        const lastAudienceActionFlow = new AudienceActionIframeFlow(
+          this.deps_,
+          params
+        );
+        this.setLastAudienceActionFlow(lastAudienceActionFlow);
+        lastAudienceActionFlow.start();
+      };
```

Comment:
Consider extracting the common logic between the two blocks, it may improve clarity by highlighting what is different in the two branches.

Reply:
Done.